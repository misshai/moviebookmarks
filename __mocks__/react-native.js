
const rn = require('react-native');

jest.mock('Linking', () => {
    return {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        openURL: jest.fn(),
        canOpenURL: jest.fn(),
        getInitialURL: jest.fn(),
    }
});

jest.mock('../src/componentes/FilmForm', () => {
    const RealComponent = require.requireActual('View');
    const React = require('React');
    class FilmForm extends React.Component {
        render() {

            return React.createElement('View', this.props, this.props.children);

        }
    }
    return FilmForm;
});

jest.mock('react-native-collapsible/Accordion', ()=>{
    const RealComponent = require.requireActual('View');
    const React = require('React');
    class Accordion extends React.Component {
        render() {

            return React.createElement('View', this.props, this.props.children);

        }
    }
    return Accordion;
});



jest.unmock('redux-mock-store');
jest.unmock('redux-thunk');
module.exports = rn;