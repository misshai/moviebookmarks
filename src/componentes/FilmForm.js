import React, { Component } from 'react';
import { View, Text, Picker } from 'react-native';
import { connect } from 'react-redux';
import { filmUpdate } from '../actions';
import { CardSection, Input } from './common';

class FilmForm extends Component {
    render() {
        return (
            <View>
                <CardSection>
                    <Input
                        label="Name of film"
                        placeholder="Big bang teory"
                        value={this.props.name}
                        onChangeText={value => this.props.filmUpdate({ prop: 'name', value })}
                    />
                </CardSection>

                <CardSection>
                    <Input
                        label="Number of seasons"
                        placeholder="4"
                        value={String(this.props.numberOfSeasons)}
                        onChangeText={value => this.props.filmUpdate({ prop: 'numberOfSeasons', value })}
                    />
                </CardSection>

                <CardSection>
                    <Input
                        label="Number of series per season"
                        placeholder="10"
                        value={String(this.props.numberOfSeries)}
                        onChangeText={value => this.props.filmUpdate({ prop: 'numberOfSeries', value })}
                    />
                </CardSection>



            </View>
        );
    }
}

const styles = {
    pickerTextStyle: {
        fontSize: 18,
        paddingLeft: 20
    }
};

const mapStateToProps = (state) => {
    const { name, numberOfSeasons, numberOfSeries } = state.filmForm;

    return { name, numberOfSeasons, numberOfSeries };
};

export default connect(mapStateToProps, { filmUpdate })(FilmForm);
