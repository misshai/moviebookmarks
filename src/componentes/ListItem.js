import React, { Component } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { CardSection } from './common';

class ListItem extends Component {
    onRowPress() {
        Actions.editFilm({film: this.props.film});
    }

    render() {
        const { name ,lastWatchedSeason, lastWatchedSerie, hasWatched} = this.props.film;
        const {container ,titleStyle, lastWatchedStyle } = styles;
        return (
            <TouchableWithoutFeedback onPress={this.onRowPress.bind(this)}>
                <View>
                    <CardSection style={container}>
                        <Text style={titleStyle}>
                            {name}
                        </Text>
                        {this.renderLastWatched()}

                    </CardSection>
                </View>
            </TouchableWithoutFeedback>
        );
    }

    renderLastWatched() {
        const { lastWatchedSeason, lastWatchedSerie, hasWatched} = this.props.film;
        const { lastWatchedStyle } = styles;
        if (hasWatched)
            return (
                <Text style={lastWatchedStyle}>
                    Last watched {lastWatchedSeason}:{lastWatchedSerie}
                </Text>);
        return (<Text style={lastWatchedStyle}>
            Not watched
        </Text>);
    }
}

const styles = {
    titleStyle: {
        fontSize: 18,
        paddingLeft: 15
    },
    lastWatchedStyle: {
        fontSize: 13,
    },
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    }
};

export default ListItem;
