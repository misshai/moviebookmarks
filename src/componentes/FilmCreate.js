import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Keyboard, TouchableWithoutFeedback} from 'react-native'
import { filmCreate } from '../actions';
import { Card, CardSection, Button,DismissKeyboardHOC } from './common';
import FilmForm from './FilmForm';

export class FilmCreate extends Component {
    onButtonPress() {
        const { name, numberOfSeasons, numberOfSeries } = this.props;
        Keyboard.dismiss();
        this.props.filmCreate({name, numberOfSeasons, numberOfSeries});
    }

    onFormPress() {
        Keyboard.dismiss();
    }

    render() {
        return (
                <Card>
                    <FilmForm {...this.props} />
                    <CardSection>
                        <Button onPress={this.onButtonPress.bind(this)}>
                            Create
                        </Button>
                    </CardSection>
                </Card>
        );
    }
}
const mapStateToProps = (state) => {
    const { name, numberOfSeries, numberOfSeasons } = state.filmForm;

    return {name, numberOfSeries, numberOfSeasons};
};

export default connect(mapStateToProps, {filmCreate})(FilmCreate);
