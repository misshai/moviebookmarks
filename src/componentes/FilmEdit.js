import _ from 'lodash';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Text, View, StyleSheet, ScrollView } from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';
import CheckBox from 'react-native-checkbox';
import FilmForm from './FilmForm';
import { filmUpdate, filmSave, filmDelete, formClosed } from '../actions';
import { Card, CardSection, Button, Confirm } from './common';

export class FilmEdit extends Component {
    state = {showModal: false};

    componentWillMount() {
        _.each(this.props.film, (value, prop) => {
            this.props.filmUpdate({prop, value});
        });
    }

    componentWillUnmount() {
        this.props.formClosed();
    }

    onButtonPress() {
        const { name, numberOfSeasons, numberOfSeries, seasons } = this.props;

        this.props.filmSave({name, numberOfSeasons, numberOfSeries, seasons, uid: this.props.film.uid});
    }


    onAccept() {
        const { uid } = this.props.film;

        this.props.filmDelete({uid});
    }

    onDecline() {
        this.setState({showModal: false});
    }

    renderHeader(section, i) {

        return (
            <View style={styles.header}>
                <Text style={styles.headerText}>Season {(i + 1).toString()}</Text>
            </View>
        );
    }

    renderContent(section, seasonNumber) {
        return (<View style={styles.content}>
            {section.map((value, serieNumber)=> {
                return (

                    <CheckBox
                        key={serieNumber}
                        label={(serieNumber+1).toString()}
                        checked={value.watched}
                        onChange={(checked)=>{
                            let seasonsMod = [...this.props.seasons];
                            seasonsMod[seasonNumber][serieNumber].watched = !checked;
                            this.props.filmUpdate({ prop: 'seasons', value:seasonsMod })
                                 }}
                    />)
            })}
        </View>);
    }


    render() {
        return (
            <ScrollView>
                <Card style={styles.container}>
                    <FilmForm />
                    <Accordion
                        sections={this.props.seasons}
                        renderHeader={this.renderHeader}
                        renderContent={this.renderContent.bind(this)}
                    />
                    <CardSection>
                        <Button onPress={this.onButtonPress.bind(this)}>
                            Save Changes
                        </Button>
                    </CardSection>


                    <CardSection>
                        <Button onPress={() => this.setState({ showModal: !this.state.showModal })}>
                            Delete Film
                        </Button>
                    </CardSection>

                    <Confirm
                        visible={this.state.showModal}
                        onAccept={this.onAccept.bind(this)}
                        onDecline={this.onDecline.bind(this)}
                    >
                        Are you sure you want to delete this film?
                    </Confirm>
                </Card>
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,

    },
    title: {
        textAlign: 'center',
        fontSize: 22,
        fontWeight: '300',
        marginBottom: 20,
    },
    header: {
        backgroundColor: '#F5FCFF',
        padding: 10,
    },
    headerText: {
        textAlign: 'center',
        fontSize: 16,
        fontWeight: '500',
    },
    content: {
        padding: 20,
        backgroundColor: '#fff',
        flexDirection: 'row',
        justifyContent: 'space-between',
        flexWrap: 'wrap',
    }
});

const mapStateToProps = (state) => {
    const { name, numberOfSeasons, numberOfSeries, seasons } = state.filmForm;

    return {name, numberOfSeasons, numberOfSeries, seasons};
};

export default connect(mapStateToProps, {
    filmUpdate, filmSave, filmDelete, formClosed
})(FilmEdit);
