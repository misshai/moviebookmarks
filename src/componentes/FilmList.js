import _ from 'lodash';
import React, { Component } from 'react';
import { ListView, Text } from 'react-native';
import { connect } from 'react-redux';
import { filmsFetch } from '../actions';
import ListItem from './ListItem';

class FilmList extends Component {
    componentWillMount() {
        this.props.filmsFetch();

        this.createDataSource(this.props);
    }

    componentWillReceiveProps(nextProps) {
        // nextProps are the next set of props that this component
        // will be rendered with
        // this.props is still the old set of props

        this.createDataSource(nextProps);
    }

    createDataSource({ films }) {
        const ds = new ListView.DataSource({
            rowHasChanged: (r1, r2) => r1 !== r2
        });

        this.dataSource = ds.cloneWithRows(films);
    }

    renderRow(film) {
        let filmItem = {...film};

        let {lastWatchedSerie,lastWatchedSeason,hasWatched} = this.getLastWatched(film.seasons);
        filmItem.numberOfSeasons = film.seasons.length;
        filmItem.numberOfSeries = film.seasons[0].length;
        filmItem.lastWatchedSerie = lastWatchedSerie;
        filmItem.lastWatchedSeason = lastWatchedSeason;
        filmItem.hasWatched = hasWatched;
        return <ListItem film={filmItem}/>;
    }

    getLastWatched(seasons) {
        let lastWatchedSerie = 0,
            lastWatchedSeason = 0,
            hasWatched = false;
        for (let i = seasons.length - 1; i >= 0; i--) {
            for (let j = seasons[i].length - 1; j >= 0; j--) {
                if (seasons[i][j].watched) {
                    hasWatched = true;
                    lastWatchedSeason = i+1;
                    lastWatchedSerie = j+1;
                    return {lastWatchedSerie, lastWatchedSeason, hasWatched};
                }
            }
        }
        return {lastWatchedSerie, lastWatchedSeason, hasWatched};
    }


    render() {
        return (
            <ListView
                enableEmptySections
                dataSource={this.dataSource}
                renderRow={this.renderRow.bind(this)}
            />
        );
    }
}


const mapStateToProps = state => {
    const films = _.map(state.films, (val, uid) => {
        return {...val, uid};
    });

    return {films};
};

export default connect(mapStateToProps, {filmsFetch})(FilmList);

