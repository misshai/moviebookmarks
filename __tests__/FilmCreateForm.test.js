import 'react-native';
import React from 'react';
import {FilmCreate} from '../src/componentes/FilmCreate';
import renderer from 'react-test-renderer';



test('Film Create Form', () => {
    const component = renderer.create(
        <FilmCreate />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});