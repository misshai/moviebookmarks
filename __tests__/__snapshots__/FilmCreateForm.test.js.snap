// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Create Form 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#ddd",
        "borderRadius": 2,
        "borderWidth": 1,
        "elevation": 1,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 2,
      },
      undefined,
    ]
  }
  testID={undefined}
>
  <View />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Create
      </Text>
    </View>
  </View>
</View>
`;
